applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -37,13 +37,29 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_USB_STATE_LOG_LEVEL);
 #define CONFIG_USB_HID_PROTOCOL_CODE -1
 #endif
 
+#define USB_HID_SELECTIVE_SUB	\
+	IS_ENABLED(CONFIG_DESKTOP_USB_SELECTIVE_HID_SUBSCRIPTION)
+
+#if USB_HID_SELECTIVE_SUB
+  #include "usb_state_def.h"
+#else
+  static const uint16_t usb_hid_selected_report_bitmask[0];
+  #if (CONFIG_DESKTOP_HID_STATE_ENABLE) && (CONFIG_USB_HID_DEVICE_COUNT > 1)
```

Comment:
avoid cross referring to hid_state here (as hid_state already refers to USB)
better to move this to hid_state.c and even better to Kconfig.hid_state

Reply:
Maybe I could switch to CONFIG_BT_PERIPHERAL here or CONFIG_DESKTOP_HID_DONGLE ? The check is actually a part of usb_state.c module. For this reason I would prefer not to move to hid_statec.

Reply:
if it cannot be moved then remove it
actually it should be gone when we keep single configuration (remove CONFIG_DESKTOP_USB_SELECTIVE_HID_SUBSCRIPTION)

Reply:
I will remake providing default configuration to keep things simpler (that could reduce code complexity). I will use util_listify macro for that purpose.